src/nlug.ts

```diff
@@ -0,0 +1,94 @@
+import { Move } from "./types";
+
+interface NLUMapping {
+  [index: string]: Move;
+}
+type NLGMapping = [Move, string][];
+
+const nluMapping: NLUMapping = {
+  "Create an appointment": {
+    "type": "request",
+    "content": "create_appointment"
+  },
+  "What's your favorite food?": {
+    type: "ask",
+    content: (x) => `favorite_food ${x}`,
+  },
+  Pizza: {
+    type: "answer",
+    content: "pizza",
+  },
+};
+const nlgMapping: NLGMapping = [
+  [
+    {
+      "type": "ask",
+      "content": {
```

Comment:
So far I express questions like functions:
      
       const q = (x) => `meeting person ${x}`

This would make resolving QUDs pretty straightforward: `q("Vlad")` evaluates to `meeting_person Vlad`. But maybe there is a better way with js objects.

Reply:
Yes, representing questions as functions is attractive. But representing propositions as strings ("meeting_person vlad") might become problematic, when such strings will need to be parsed, e.g. to identify the argument. How about something like:

`const q = (x) => {"type": "proposition", "content": {"predicate": "meeting_person", "argument": x}}`

Reply:
Looks good to me! (it should be either with parenthesis -- `const q => ({ ... })` or return -- `const q => {return { ... }}`, or some people recommend against this syntax -- `function q(x) => {return { ... }}`).